rasa/utils/tensorflow/models.py

```diff
@@ -35,6 +48,37 @@ def __init__(self, random_seed: Optional[int] = None, **kwargs) -> None:
 
         self.random_seed = random_seed
 
+        self.train_summary_writer = None
+        self.test_summary_writer = None
+        self.model_summary_file = None
+        self.tensorboard_log_on_epochs = True
+
+        self._set_up_tensorboard_writer(tensorboard_log_level, tensorboard_log_dir)
+
+    def _set_up_tensorboard_writer(
+        self, tensorboard_log_level: Text, tensorboard_log_dir: Optional[Text] = None
+    ) -> None:
+        if tensorboard_log_dir is not None:
+            if tensorboard_log_level not in TENSORBOARD_LOG_LEVELS:
+                raise ValueError(
+                    f"Provided '{TENSORBOARD_LOG_LEVEL}' ('{tensorboard_log_level}') "
+                    f"is invalid! Valid values are: {TENSORBOARD_LOG_LEVELS}"
+                )
+
+            self.tensorboard_log_on_epochs = tensorboard_log_level == "epoch"
+
+            current_time = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+
+            train_log_dir = f"{tensorboard_log_dir}/{current_time}/train"
+            test_log_dir = f"{tensorboard_log_dir}/{current_time}/test"
+
+            self.train_summary_writer = tf.summary.create_file_writer(train_log_dir)
+            self.test_summary_writer = tf.summary.create_file_writer(test_log_dir)
+
+            self.model_summary_file = (
```

Comment:
Should we add the component name as a subfolder? Incase the same log directory is passed to all components, then the log directory becomes messy with just folders named after timestamps.